tubemq-core/src/main/java/org/apache/tubemq/corebase/utils/DataConverterUtil.java

```diff
@@ -233,8 +233,17 @@
                     payloadDataLen -= attrLen;
                 }
             }
-            final byte[] payload = new byte[payloadDataLen];
+            byte[] payload = new byte[payloadDataLen];
             System.arraycopy(payloadData.array(), readPos, payload, 0, payloadDataLen);
+            String compressionType = "";
```

Comment:
The extraction of the compression type value needs to be performed by calling the decoding function of the Message attribute. With the iteration of the version, it may be changed later, and it is more appropriate to process it by the method